libdns/netcup: Upgrade and adapt to libdns v1.0.0

oltdaniel opened the pull request:
Since the release of libdns `v1.0.0`, caddy also upgraded in their version `v2.10.0` to the new stable release of libdns. This caused the caddy-dns package for netcup to break, as mentioned in https://github.com/caddy-dns/netcup/issues/9 .

I've updated the libdns package to `v1.0.0` and made the required changes to pass the Tests. Slight modifications to the tests were required but should result in the same behavior. 

The main concern is the missing ID field in the new libdns Record Structs, which requires the library to choose the first found record to be chosen. However, this was already implemented before this change and should work as expected.

> As this is my first work with the `libdns` library, feel free to do a thorough review of my changes. 

Fixes #2

oltdaniel committed:
libdns/netcup: Upgrade and adapt to libdns v1.0.0

* Update go.mod - Bump libdns to v1.0.0

* Update util.go, provider_test.go - Adapt for changes in libdns v1.0.0

* Go mod tidy and mention ID field change in documentation

oltdaniel commented:
@mholt Just a quick ping, to check if/how we can get this merged to continue fixing the issues in the projects dependent on this package.

mholt commented:
If the maintainer doesn't show up to do a review in the next few days, feel free to ping me again and I can step in. (Including possibly adding another maintainer?)

I obviously don't use netcup so can't really review this fully, so I just did a quick scan and one thing I did notice is that, I think, `libdns.RR` is being returned, instead of the concrete struct type correlation with the RR-type. Note the libdns godoc:

> Implementations of the libdns interfaces should accept as input any [Record](https://pkg.go.dev/github.com/libdns/libdns#Record) value, and should return as output the concrete struct types that implement the [Record](https://pkg.go.dev/github.com/libdns/libdns#Record) interface (i.e. [Address](https://pkg.go.dev/github.com/libdns/libdns#Address), [TXT](https://pkg.go.dev/github.com/libdns/libdns#TXT), [ServiceBinding](https://pkg.go.dev/github.com/libdns/libdns#ServiceBinding), etc). This is important to ensure the provider libraries are robust and also predictable: callers can reliably type-switch on the output to immediately access structured data about each record without the possibility of errors. Returned values should be of types defined by this package to make type-assertions reliable.

oltdaniel committed:
libdns/netcup: Return record specific struct rather than only RR structs

As mentioned by mholt in
https://github.com/libdns/netcup/pull/3#issuecomment-2899367821 it is
better to return the correct record struct rather than only RR structs.
This commit fixes this behavior.

oltdaniel commented:
@mholt Thanks for the feedback. I implemented the changes by returning the parsed RR output with the correct struct type. Additionally, errors when invalid records occur are now directly returned, but I haven‘t wrapped them in this case.

It also looks like, there isn‘t an active maintainer.

ButlerMiles commented:
@mholt Could you take a look at the MR and then merge it if possible?

mholt commented:
You got it. If anyone would like to help maintain this repository, let me know and I'll add you.

oltdaniel commented:
@mholt Thanks for merging. Feel free to add me as new contributor. I think a new v1.0.0 tag in this repo might be nice, just like the other caddy-dns repos have done it with this change.

mholt commented:
@oltdaniel Done, thank you!! You also have permission to push tags.
